Clientside dev

ashish-compro opened the pull request:
Code Review feedback incorporated.

aman-compro committed:
Change the Name of the Factory Variable in Input Factory Service
Used ViewCount as Variable name in step input area.
used selectedViewNumber instead of selectedView as variable name
Used itemConfig as variable name instead of ModelInput in Input Factory service

aman-compro committed:
In UI Config model, type of items changed to node.In View Input area Component, changed the type from any to the model type.

aman-compro committed:
Moved Dynamic components, input factory service and ui config model to separate module

ashish-compro committed:
Implementing a base class for all the dynamic components.

ashish-compro committed:
Renaming group component to panel component.

aman-compro committed:
Documentaion provided for INput Factory Service

ashish-compro committed:
Changing the name of the config model class from node to itemSchema.

dhirendra-singh81 commented on client/app/shared/input-factory.service.ts:
```diff
@@ -11,28 +11,39 @@ export class InputFactoryService {
-    // add inputs first !! otherwise component/template crashes ..
-    comp.instance["data"] = modelInput;
+    // add Data to the instance of the Component.
+    comp.instance["compConfig"] = itemConfig;
```
Add a setter fn here

dhirendra-singh81 commented on client/app/shared/panel/panel.component.ts:
```diff
@@ -0,0 +1,33 @@
+    this.labelConfig.rendererProperties.text = this.compConfig.label;
+    this.labelConfig.rendererProperties.type = 'PanelHeading';
+    this.$el.find('.widget').widgster();
+    for (let item of this.compConfig.items) {
```
Put it in a function or maybe we define an interface for a non-leaf component

dhirendra-singh81 reviewed (changes requested):
1. Add non-leaf and derive appropriate component from it 
2. Add a setter fn in input factory service comp

ashish-compro committed:
Implementing the setter function for setting the data of the dynamic components.

vsingb3 commented on client/app/shared/input-factory.service.ts:
```diff
@@ -11,28 +11,39 @@ export class InputFactoryService {
 
-  public createComp (vCref: ViewContainerRef, modelInput){
-     let factory = this.componentFactoryResolver.resolveComponentFactory(this.DynamicCompMap(modelInput.itemRenderer));
+  public createComp (vCref: ViewContainerRef, itemConfig){
```
vCRef

vsingb3 commented on client/app/shared/input-factory.service.ts:
```diff
@@ -11,28 +11,39 @@ export class InputFactoryService {
+  public createComp (vCref: ViewContainerRef, itemConfig){
 
+    /**
+     * [resolveComponentFactory - We create a factory out of the component we want to create]
```
Get the factory object to create the component

aman-compro committed:
Dropzone Cmponent and Wrapper Component Added. Updated TabComponent

aman-compro committed:
Merge branch 'clientside-dev' of https://github.com/Compro-Single-Step/SIMS-Builder into clientside-dev

aman-compro committed:
Added Static HTML for Header and Footer in StepBuilder Component

aman-compro committed:
Minor Footer UI Fix

ashish-compro committed:
Adding a new hierarchy class between base component and the non-leaf components.

ashish-compro committed:
Adding comments and changing a variable name as per feedback.

ashish-compro committed:
Deriving wrapper component from elements container component.

vsingb3 commented on client/app/shared/select/select.component.html:
```diff
@@ -0,0 +1,4 @@
+<div><app-label [compConfig]="labelConfig"></app-label></div>
```
Id should be unique for every instance of this component.

Name should be generic

aman-compro committed:
Added View Navigator HTML, Chnaged CSS TO SCSS

ashish-compro committed:
Committing the functionality of view changing through view navigator component.

aman-compro committed:
Fixed View Navigator bug

Commit:
Commiting code for Global CSS

aman-compro committed:
Global SCSS Integrated

aman-compro committed:
Added Dropzone scss (Dashed Border and Margin)

prateekbhutani committed:
Adding Skill Manager Service

No regression impact.

Commit:
Removing extra files from Global Css

aman-compro committed:
Removed UI Config Button

Commit:
Removing extra files

prateekbhutani committed:
test

prateekbhutani committed:
Updating Function Name

prateekbhutani committed:
Skill Manager Service and Dropzone Update

1. UIConfig Model - Updating model stored in Step Builder and corresponding schema.
2. Using Component Config in dropzone.
3. Adding second function in Skill Manager service.

aman-compro committed:
Add Height and Width option to dropzone

aman-compro committed:
Resolve conflict

aman-compro committed:
change dimension values in stepbuilder

prateekbhutani committed:
Getting Init Doc JSON from Dropzone

Regression not expected.

prateekbhutani committed:
Moving property accessing to getNestedObject function

Regression not expected.

prateekbhutani committed:
Updating addSheetNamesToDropdown Function

No Regression Impact

ashish-compro committed:
Committing the model changes required for two way binding.

prateekbhutani committed:
Adding Rules to UI-Config JSON.
Updating Sheetin Action type.
Accepting Data Model into Skill Manager Service from UIConfig.model

Regression Not Expected.

aman-compro committed:
Add Custom Scroll bar,fix position of stepbuilder components' header and footer, add visited css to view navigator

aman-compro committed:
Resolved conflicts

tanujaggarwal committed:
Removing duplicate loader for font files
This duplicate loader was making fiont files corrupt.
This loader was added unintentionally while adding global css.

tanujaggarwal committed:
Removing duplicate loader for font files
This duplicate loader was making fiont files corrupt.
This loader was added unintentionally while adding global css.

aman-compro committed:
Remove comments and console statements

aman-compro committed:
Removed Fetch Config function and added flex wrap css

aman-compro committed:
Merge branch 'develop' of https://github.com/Compro-Single-Step/SIMS-Builder into clientside-dev

# Conflicts:
#	package.json

aman-compro committed:
Update uiconfig file on server, Fetch uiconfig from server

Commit:
Merge branch 'develop' into clientside-dev

dhirendra-singh81 reviewed (changes requested):
Unapproving

Commit:
Merge branch 'develop' into clientside-dev

aman-compro committed:
Fix the get API URL

aditi-kapur committed:
Merge pull request #21 from Compro-Single-Step/getuiconfig_api_correction

Fixed the get API URL

dhirendra-singh81 reviewed (approved):
Logically reviewed. Code refinements still needed.
